app/src/main/java/com/codepath/daggerexample/MainActivity.java

```diff
@@ -39,24 +45,31 @@ protected void onCreate(Bundle savedInstanceState) {
         fab.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(final View view) {
-                mGitHubApi.getRepos("codepath", new GitHubApi.ResponseHandler() {
-                    @Override
-                    public void onResponse(Object data) {
-                        Log.i("(MainActivity.java:45)",data.toString());
-                        Snackbar.make(view,"Data retrieved", Snackbar.LENGTH_LONG)
-                                .setAction("Action",null).show();
-                    }
-
-                    @Override
-                    public void onFailure() {
-
-                    }
-                });
-            }
-        });
+                    Call<ArrayList<Repository>> call = mGitHubApiInterface.getRepository("codepath");
+
+                    call.enqueue(new Callback<ArrayList<Repository>>() {
+                        @Override
+                        public void onResponse(Response<ArrayList<Repository>> response, Retrofit retrofit) {
+                            if (response.isSuccess()) {
+                                Log.i("DEBUG", response.body().toString());
+                                Snackbar.make(view,"Data retrieved", Snackbar.LENGTH_LONG)
+                                        .setAction("Action",null).show();
+                            } else {
+                                Log.i("ERROR", String.valueOf(response.code()));
+                            }
+
+                        }
+
+                        @Override
+                        public void onFailure(Throwable t) {
+
+                        }
+                    });
+                }
+
+            });
 
         ((MyApp) getApplication()).getGitHubComponent().inject(this);
-        mGitHubApi = new GitHubApi(mRetrofit);
```

Comment:
The issue is here.  You are avoiding dependency injection and inserting Retrofit into here when it should be a provider call:

https://github.com/codepath/android_guides/wiki/Dependency-Injection-with-Dagger-2#component-dependencies

```java
@Module
public class GitHubModule {

    public interface GitHubApiInterface {
      @GET("/org/{orgName}/repos")
      Call<ArrayList<Repository>> getRepository(@Path("orgName") String orgName);
    }

    @Provides
    @UserScope // needs to be consistent with the component scope
    public GitHubApiInterface providesGitHubInterface(Retrofit retrofit) {
        return retrofit.create(GitHubApiInterface.class);
    }
}

```